File: 7KYU/Battle of the characters (Easy)/solution.rb
def score(x)
  x.chars.map {|x| x.codepoints[0]-64 }.reduce {|acc,i| acc+=i}
end

def battle(x, y)
  x_score = score(x)
  y_score = score(y)
  
  if x_score > y_score
    x
  elsif x_score < y_score
    y
  else
    "Tie!"
  end
end

File: 7KYU/Battle of the characters (Easy)/solution.c
const int score(const char* str) {
  int score = 0;
  
  for(int i = 0; str[i] != '\0'; i++) {
    score += str[i]-64;
  }
  return score;
}

const char *battle (const char *group_1, const char *group_2)
{
  int g1_score = score(group_1);
  int g2_score = score(group_2);
  
  if(g1_score > g2_score) {
    return group_1;
  } else if (g1_score < g2_score) {
    return group_2;
  } else {
    return  "Tie!";
  }
}